src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1949,10 +1949,10 @@ bool computeIsGeneratedCode()
                 // First check for explicit user configuration for generated code.
                 //     generated_code = true | false
                 var options = AnalyzerExecutor.AnalyzerOptions.AnalyzerConfigOptionsProvider.GetOptions(tree);
-                if (options.TryGetValue("generated_code", out string optionValue) &&
-                    bool.TryParse(optionValue, out var boolValue))
+                var isUserConfiguredGeneratedCode = GeneratedCodeUtilities.GetIsGeneratedCodeFromOptions(options);
+                if (isUserConfiguredGeneratedCode.HasValue)
```

Comment:
>if (isUserConfiguredGeneratedCode.HasValue) [](start = 16, length = 43)

It feels like this should use `??`, although that would mean moving comments around.
```
return GetIsGeneratedCodeFromOptions(options) ??
    _isGeneratedCode(...);
```